demo-app/client/generated/common/formArrayExtended.ts

```diff
@@ -0,0 +1,23 @@
+/* tslint:disable:max-line-length */
+/**
+ * Test Swagger
+ * v1
+ * example.com/api-base-path
+ */
+
+import {AbstractControl, FormArray} from '@angular/forms';
+
+type ControlFactory = () => AbstractControl;
+
+/** Extends FormArray so it contains definitions of items for further creation */
+export class FormArrayExtended extends FormArray {
+  constructor(public createControl: ControlFactory, ...rest: any[]) {
+    super([], ...rest);
+  }
+
+  setValue(value: any[], options: {onlySelf?: boolean, emitEvent?: boolean} = {}): void {
+    while (value.length < this.controls.length) this.removeAt(this.length - 1);
```

Comment:
it could be beneficial to have an optional additional argument `removeAtPosition` with default value, so that you can choose at which position an element will be deleted. What do you think?

Reply:
And also maybe `addToPosition`. Let's do it in the next step once we need exactly what's needed based on real-life usage.

Reply:
Btw, these `...atPosition` methods are already present (`.insert(), .removeAt()`). And it should also be possible to modify FormArray's items directly via its `.controls` array.